filter/filter_test.go

```diff
@@ -160,10 +161,12 @@ var _ = Describe("Filter", func() {
 			t("foo == bar", "foo", "==", "bar")
 		})
 
-		It("Should fail for facts in the wrong format", func() {
-			pf, err := ParseFactFilterString("foo")
-			Expect(err).To(MatchError("could not parse fact foo it does not appear to be in a valid format"))
-			Expect(pf).To(BeNil())
+		It("Should fail on invalid fact filters", func() {
```

Comment:
looks great, lets add a few tests against these new kinds of filters that would have triggered the bug and this one is done!